src/autoscaler.ts

```diff
@@ -167,66 +177,79 @@ export default class AutoscaleProcessor {
-                            (count < group.scalingOptions.maxDesired &&
-                                availableForPeriod < group.scalingOptions.scaleUpThreshold) ||
-                            count < group.scalingOptions.minDesired
-                        );
-                    })
-                    .reduce((previousValue, currentValue) => {
-                        return previousValue && currentValue;
-                    });
+                break;
             case 'JVB':
-                // @TODO: implement scale up algorithm for JVB autoscaling
+                // in the case of JVB scale up only if value (average stress level) is above or equal to threshhold
+                return (
+                    (count < group.scalingOptions.maxDesired && value >= group.scalingOptions.scaleUpThreshold) ||
+                    count < group.scalingOptions.minDesired
+                );
                 break;
         }
         return false;
     }
 
-    async evalScaleDownConditionForAllPeriods(
-        ctx: Context,
-        scaleMetrics: Array<number>,
-        count: number,
-        group: InstanceGroup,
-    ): Promise<boolean> {
+    private scaleDownChoice(group: InstanceGroup, count: number, value: number): boolean {
         switch (group.type) {
             case 'jibri':
-                ctx.logger.info(
-                    `[AutoScaler] Evaluating jibri scale down for group ${group.name} with ${count} instances and current desired count ${group.scalingOptions.desiredCount}`,
-                    { scaleMetrics },
-                );
-
-                return scaleMetrics
-                    .slice(0, group.scalingOptions.scaleDownPeriodsCount)
-                    .map((availableForPeriod) => {
-                        return (
-                            count > group.scalingOptions.minDesired &&
-                            availableForPeriod > group.scalingOptions.scaleDownThreshold
-                        );
-                    })
-                    .reduce((previousValue, currentValue) => {
-                        return previousValue && currentValue;
-                    });
+                // in the jibri case only scale up if value (available count) is above threshold
+                return count > group.scalingOptions.minDesired && value > group.scalingOptions.scaleDownThreshold;
+                break;
             case 'JVB':
-                // @TODO: implement scale up algorithm for JVB autoscaling
+                // in the case of JVB scale down only if value (average stress level) is below threshhold
+                return count > group.scalingOptions.minDesired && value < group.scalingOptions.scaleDownThreshold;
                 break;
         }
+
         return false;
     }
+
+    private evalScaleConditionForAllPeriods(
+        ctx: Context,
+        scaleMetrics: Array<number>,
+        count: number,
+        group: InstanceGroup,
+        scaleUpFlag: boolean,
```

Comment:
Perhaps just send scaleDirection (up/down) as a parameter, instead of scaleUpFlag.